blocklist-client/src/client/client.rs

```diff
@@ -1,20 +1,316 @@
+
+    // If registration is successful, proceed to check the address
+    let assessed_response = get_risk_assessment(client, config, address).await?;
+    debug!(
+        "Received risk assessment response: {} for address: {}",
+        register_response.address, register_response.address
+    );
+
+    let blocklist_status = BlocklistStatus {
+        // `is_blocklisted` is set to true if risk is Severe
+        is_blocklisted: matches!(assessed_response, RiskSeverity::Severe),
+        severity: assessed_response.to_string(),
+        // `accept` is set to false if severity is Severe
+        accept: !matches!(assessed_response, RiskSeverity::Severe),
+    };
+
+    Ok(blocklist_status)
+}
+
+/// Evaluates the HTTP response from an API request and translates HTTP status codes into application-specific errors.
+async fn check_api_response(response: Response) -> Result<Response, Error> {
+    match response.status() {
+        StatusCode::OK | StatusCode::CREATED => Ok(response),
+        StatusCode::BAD_REQUEST => Err(Error::HttpRequestError(
+            response.status(),
+            "Bad request - Invalid parameters or data".to_string(),
+        )),
+        StatusCode::FORBIDDEN => Err(Error::Unauthorized),
+        StatusCode::NOT_FOUND => Err(Error::NotFound),
+        StatusCode::NOT_ACCEPTABLE => Err(Error::NotAcceptable),
+        StatusCode::CONFLICT => Err(Error::Conflict),
+        StatusCode::INTERNAL_SERVER_ERROR => Err(Error::InternalServerError),
+        StatusCode::SERVICE_UNAVAILABLE => Err(Error::ServiceUnavailable),
+        StatusCode::REQUEST_TIMEOUT => Err(Error::RequestTimeout),
+        status => Err(Error::HttpRequestError(
+            status,
+            "Unhandled status code".to_string(),
+        )),
+    }
+}
+
+fn register_address_path(base_url: &str) -> String {
+    format!("{}{}", base_url, API_BASE_PATH)
+}
+
+fn risk_assessment_path(base_url: &str, address: &str) -> String {
+    format!("{}{}/{}", base_url, API_BASE_PATH, address)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::common::RiskSeverity::{Low, Severe};
+    use mockito::{mock, server_url, Mock};
+
+    const TEST_ADDRESS: &str = "test_address";
+    const ADDRESS_REGISTRATION_BODY: &str = r#"{"address": "test_address"}"#;
+
+    // Setup function for common client and configuration
+    fn setup_client() -> (Client, RiskAnalysisConfig) {
+        let client = Client::new();
+        let config = RiskAnalysisConfig {
+            api_url: server_url(),
+            api_key: "dummy_api_key".to_string(),
+        };
+        (client, config)
+    }
+
+    // Helper function to setup a mock API response
+    fn setup_mock(method: &str, path: &str, status: u16, body: &str) -> Mock {
+        return mock(method, path)
+            .with_status(status.into())
+            .with_header("content-type", "application/json")
+            .with_body(body)
+            .create();
```

Comment:
Nice. I've used `httpmock` for this kind of functionality, but `mockito` seems to have a bit of a nicer API. Good to know.